### PR TITLE
An 4097/dim idl

### PR DESCRIPTION
**Pull request by tarikceric**
Add DIM table containing information on IDLs/programs that are being decoded.

### CODE REVIEW

**Review by desmond-hui — CHANGES_REQUESTED** [review]

File: models/silver/idls/silver__idl.sql [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    SELECT
+        A.earliest_decoded_block,
+        A.program_id,
+        C.idl,
+        C.idl_hash
+    FROM
+        idls A
+        LEFT JOIN {{ ref('streamline__idls_history') }}
+        b
+        ON A.program_id = b.program_id
```
[the remark this answers is not in the document]
Comment (tarikceric) [reply models/silver/idls/silver__idl.sql:27]: removed

File: models/silver/idls/silver__idl.sql [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{{ config(
+    materialized = 'incremental',
+    unique_key = "program_id",
+    tags = ['scheduled_non_core']
```
[the remark this answers is not in the document]
Comment (tarikceric) [reply models/silver/idls/silver__idl.sql:4]: added

File: models/silver/idls/silver__idl.sql [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+) }}
+
+WITH idls AS (
```
[the remark this answers is not in the document]
Comment (tarikceric) [reply models/silver/idls/silver__idl.sql:7]: add incremental logic to only check decoded_instructions from last two hours, and updates only if the min(block_id) from the first cte is less than the existing earliest_decoded_block

**Review by desmond-hui — APPROVED** [review]
See comment

**Pull request merged**